http/handler.go

```diff
@@ -847,11 +850,15 @@ func forwardRequest(core *vault.Core, w http.ResponseWriter, r *http.Request) {
 // case of an error.
 func request(core *vault.Core, w http.ResponseWriter, rawReq *http.Request, r *logical.Request) (*logical.Response, bool, bool) {
 	resp, err := core.HandleRequest(rawReq.Context(), r)
-	if errwrap.Contains(err, consts.ErrStandby.Error()) {
-		respondStandby(core, w, rawReq.URL)
-		return resp, false, false
+
+	isForwardErr := func(e error) bool {
+		return errwrap.Contains(e, logical.ErrPerfStandbyPleaseForward.Error()) ||
+			errwrap.Contains(e, logical.ErrReadOnly.Error()) ||
+			errwrap.Contains(e, consts.ErrStandby.Error())
 	}
-	if err != nil && errwrap.Contains(err, logical.ErrPerfStandbyPleaseForward.Error()) {
+
+	if isForwardErr(err) || (resp != nil && isForwardErr(resp.Error())) {
```

Comment:
Ah, never mind, ignore my comment. This is just a different set of error conditions. I agree with these changes :laughing: